### DISCUSSION TITLE
Local build succeeds but vercel build fails with typescript error

### DISCUSSION THREAD

**Discussion by xixixao** [vercel/community#5509] [Help] [answered]
### Summary

I am getting a typescript inference error only on remote vercel build, not locally. I have only standard public NPM dependencies.

There's nothing interesting in the error, it complains that TypeScript thinks my argument is :any, but locally it's inferred fine. I do have 2 `tsconfig.json` files, but these should work the same locally as they do on Vercel.

I deployed successfully via `npx vercel build --prod; and npx vercel deploy --prebuilt --prod`. How can I reproduce the environment Vercel is trying to build in?

### Steps to Reproduce

Build: https://vercel.com/xixixao/saas-starter/4Y7rFRqcgou3tLMtftsEzkXCejdG
Error:

```

Failed to compile.
--
16:37:45.547 |  
16:37:45.547 | ./app/t/Notifications.tsx:39:24
16:37:45.547 | Type error: Parameter 'invite' implicitly has an 'any' type.
```

**Comment by versecafe** [comment, marked as answer]
From testing it appears this issue exists in local development too and in several different files such as `./app/t/[teamSlug]/hook.ts` the source of the issue appears to be `useQuery(api.invites.list);` throwing `type instantiation is excessively deep and possibly infinite.ts(2589)` which is preventing the type inference in every instance where this issue appears to be occurring.

**Reply by xixixao** [reply]
@versecafe Interesting, that is not what I'm seeing locally. I wonder what's different, do you have a way to reproduce the Vercel environment closely?
I can see that TypeScript is failing in the Vercel environment (on my latest deployment the typescript-aware lint is giving the wrong result). Maybe my machine is beefier and TypeScript has some perf-based limit on the deep instantiation?

**Reply by versecafe** [reply]
I'm running an M3 Macbook Pro with 36GB of RAM so I doubt it is machine specs themselves, slight chance if you are on Windows it may be causing the difference as most Unix operating systems behave similar and Vercel is running a Linux distro for builds. you could also look [here](https://www.typescriptlang.org/play?#code/C4TwDgpgBACgThAblAvFA2gOyROAaKABgIEYCAmAgZgIBYCBWAgNgIHYCAOAgTgF0A3ACghoSFACyAewAmAVwA2EADwwCAYQB8qKAG8hUKGACGIBVOMyAzgC5YwwwGMAFgEsFMhJjvrhAX2FRcGhpeSUAJSkpYABBR2BXKUwrZQkoCAAPYAhMa0lZRRVjTBACYpBNbTR9Q3QAOShXTCgAawgQKQAzSXQAIhMzC2tevj47GsMoMQg7OodJgfNLOwk+xaGrEfrBAygAoT90No7u1f7TJeHRwOmoGIUFUMK4hKSUtMzs3Kt8sKKSsolTQEAAi6SyOTy8CQWDkAFsAEa4Pg6WhVXboEEoz6Qn5YHBwFEAfig2EQuCgNl2AB9foVItEXolkqlNDS9LtagBpRrNY5dHq9FzuTw5EZjQXCjxeLZc7EQ750pTKcqAiqcqAk+6PApKJlvVJ9KWizCyvgEaGITF8NmTSmkgnzQ7805GtzSsXXERCAD0ACoA4G-VAAKIADRiEhgABkQwBlKBBwM+oSON7AKAAcxyuFcjh0qXBXzyT2Vqqg5UqAApdsZ4sy7NrSxB9SyJNofT6oMoALQ9qAyCCdXBCACUqG0Ezrr0w-mEQA) for an example of how to defer the type inference which may solve your problem.
